### PR TITLE
Clarify language on SIG PM participation

### PR DESCRIPTION
**Pull request by justaugustus**
There is mention in a few places about the role of a SIG PM representative, which isn't actually a role that needs to be staffed.
This led to some confusion on my part (https://github.com/kubernetes/community/issues/2130), as well as an attempt to update the currently outdated list (https://github.com/kubernetes/community/issues/2122).

After some discussion in SIG PM, we decided to remove the SIG PM list and cleanup / clarify the language around SIG PM participation.

Hopefully, this:
* fixes https://github.com/kubernetes/community/issues/2130
* closes https://github.com/kubernetes/community/issues/2122

/cc @calebamiles @idvoretskyi @jdumars

### CODE REVIEW

File: sig-governance.md [hunk 1]
```diff
@@ -11,12 +11,18 @@ In order to standardize Special Interest Group efforts, create maximum transpare
...
 
-## SIG roles
-- **SIG Participant**: active in one or more areas of the project; wide 
-  variety of roles are represented
-- **SIG Lead**: SIG organizer
+In addition, SIGs have the following responsibilities to SIG PM:
+* identify SIG annual roadmap
+* identify all SIG features in the current release
+* actively track / maintain SIG features within [k/features](https://github.com/kubernetes/features)
+* attend [SIG PM](sig-pm/README.md) meetings, as needed / requested
```
Comment (cblecker) [inline sig-governance.md]: nit: this link should be `/sig-pm/README.md` (note the leading slash)

File: sig-governance.md [hunk 2]
```diff
@@ -11,12 +11,18 @@ In order to standardize Special Interest Group efforts, create maximum transpare
...
+
+## SIG Roles
+
+Defining SIG Roles is a function of the SIG Charter.  
+Guidelines for drafting a SIG Charter can be found [here](committee-steering/governance/README.md).
```
Comment (cblecker) [inline sig-governance.md]: nit: this link should be `/committee-steering/governance/README.md` (note the leading slash)

**Comment by justaugustus** [comment]
@cblecker I fixed the nits. No other reviews required as @idvoretskyi approved already.
/lgtm
/hold cancel

**Comment by justaugustus** [comment]
Sorry @k8s-ci-robot; I knew that.
@idvoretskyi can you re-`/lgtm` this?

**Pull request merged**